lib/src/domain/communication/firebase_exception.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:allthenews/generated/l10n.dart';
+import 'package:allthenews/src/domain/communication/all_the_news_exception.dart';
+import 'package:flutter/material.dart';
+
+class AuthenticationException extends AllTheNewsException {
+  final String message;
+
+  AuthenticationException({@required this.message});
+}
+
+class NetworkException extends AuthenticationException {
+  NetworkException([String message]) : super(message: message ?? Strings.current.noInternetException);
```

Comment:
to są błędy domenowe, więc powinny być w pakiecie domenowym i nie powinny korzystać ze Strings, które są w warstwie prezentacji :)

Reply:
raczej wrzuciłbym je do warstwy prezentacji i pozwolił korzystać ze stringsów